src/DistributedComputations/distributed_fft_based_poisson_solver.jl

```diff
@@ -55,155 +47,159 @@ Algorithm for two-dimensional decompositions
+4. Transform along ``y``.
+5. Transpose + communicate to `storage.xfield` partitioned into `(Rx, Ry)` processes in ``(y, z)``.
+6. Transform in ``x``.
 
 At this point the three in-place forward transforms are complete, and we
-solve the Poisson equation by updating `last(storage)`.
-Then the process is reversed to obtain `first(storage)` in physical
-space and with the layout ``(z, x, y)``.
+solve the Poisson equation by updating `storage.xfield`.
+Then the process is reversed to obtain `storage.zfield` in physical
+space partitioned over ``(x, y)``.
+
+Algorithm for one-dimensional decompositions
+============================================
+
+The one-dimensional decomposition works in the same manner while skipping the transposes that
+are not required. Foe example if the domain is decomposed in ``x``, step 3 in the above algorithm
+is skipped (and the associated transposition step in the bakward transform)
 
 Restrictions
 ============
 
-The algorithm for two-dimensional decompositions requires that `Nz = size(global_grid, 3)` is larger
-than either `Rx = ranks[1]` or `Ry = ranks[2]`, where `ranks` are configured when building `Distributed`.
-If `Nz` does not satisfy this condition, we can only support a one-dimensional decomposition.
+1. Two-dimensional decomopositions:
+    - `Ny ≥ Rx` and `Ny % Rx = 0`
+    - `Nz ≥ Ry` and `Nz % Ry = 0`
+    - If the ``z`` direction is `Periodic`, also the ``y`` and the ``x`` directions must be `Periodic`
+    - If the ``y`` direction is `Periodic`, also the ``x`` direction must be `Periodic`
 
-Algorithm for one-dimensional decompositions
-============================================
-
-This algorithm requires a one-dimensional decomposition with _either_ `Rx = 1`
-_or_ `Ry = 1`, and is important to support two-dimensional transforms.
+2. One-dimensional decomposition:
+    - same as for two-dimensional decompositions with `Rx` (or `Ry`) equal to one
 
-For one-dimensional decompositions, we place the decomposed direction _last_.
-If the number of ranks is `Rh = max(Rx, Ry)`, this algorithm requires that 
-_both_ `Nx > Rh` _and_ `Ny > Rh`. The resulting flow of transposes and transforms
-is similar to the two-dimensional case. It remains somewhat of a mystery why this
-succeeds (i.e., why the last transform is correctly decomposed).
 """
-function DistributedFFTBasedPoissonSolver(global_grid, local_grid)
+function DistributedFFTBasedPoissonSolver(global_grid, local_grid, planner_flag=FFTW.PATIENT)
 
-    arch = architecture(local_grid)
-    Rx, Ry, Rz = arch.ranks
-    communicator = arch.communicator
+    validate_global_grid(global_grid)
+    FT = Complex{eltype(local_grid)}
 
+    storage = TransposableField(CenterField(local_grid), FT)
     # We don't support distributing anything in z.
+    Rz = architecture(local_grid).ranks[3]
     Rz == 1 || throw(ArgumentError("Non-singleton ranks in the vertical are not supported by DistributedFFTBasedPoissonSolver."))
```

Comment:
I don't see tests for the other requirements (such as `Nz ≥ Ry`, etc.). Should we test for those as well?

Reply:
I have added a validation before the construction of the distributed solver